### ISSUE TITLE
Bad entries created on Cloudflare DNS

### ISSUE THREAD

**Issue by QDespeisseTalan** [ibm-mas/installer#548]
Hello,

While installing MAS through CLI, I setted up a custom domain with a Cloudflare DNS.
The `suite-dns` task created multiple entries for MAS subdomains as alias of the `ocp_hostname`.
The `ocp_hostname` was defined by `TASK [ibm.mas_devops.suite_dns : cloudflare : Set ocp_hostname]` task.
These entries were causing `DNS_PROBE_FINISHED_NXDOMAIN` while trying to access MAS.
After replacing the alias of all DNS entries by the value of the `routerCanonicalHostname`, MAS was accessible.

**Comment by andrercm** [comment]
@QDespeisseTalan thanks for you feedback... can you share how do you retrieve the `routerCanonicalHostname` information? the automation is currently setting `ocp_hostname` based on cluster's ingress domain... if that's not working for all cases, then perhaps we could allow this property to be prompted and set by users so this could be overwritten when cluster's ingress is not intended.

**Comment by QDespeisseTalan** [comment]
Hi @andrercm ,
On Openshift console, Networking > Routes > open the mas-home route for example and you'll find at the bottom the `Router Canonical Hostname` that is used by Openshift for a custom DNS :
![image](https://github.com/ibm-mas/cli/assets/147660805/8737afea-2ddc-412b-be14-b83a78432900)

I found another way to correct the issue :
When you install Openshift, it requires to add two entries on the DNS (this example in my case) :
```
api.mas.servicemanageteamaca.fr      A	137.74.225.100
*.apps.mas.servicemanageteamaca.fr   A	137.74.225.101
```

The Set ocp_hostname task has defined the Openshift hostname as `apps.mas.servicemanageteamaca.fr`
Anf after, `suite-dns` adds MAS redirections to `apps.mas.servicemanageteamaca.fr` but as you can see, I don't have any redirection for `apps.mas.servicemanageteamaca.fr` so I just added the following entry to the DNS to make it work :
`apps.mas.servicemanageteamaca.fr   A	137.74.225.101
`
To sum up :
The `Set ocp-hostname` task probably works perfectly, but it should look after the DNS if an entry exists on this hostname and if not, create one to redirect it to the Ingress IP.

**Comment by andrercm** [comment]
@durera @joaopauloksn based on the feedback given by @QDespeisseTalan, one of the potential fixes to address this problem would be to change here:

from:

https://github.com/ibm-mas/ansible-devops/blob/master/ibm/mas_devops/roles/suite_dns/tasks/providers/cloudflare/main.yml#L61

```
- name: "cloudflare : Set ocp_hostname"
  set_fact:
    ocp_hostname: "{{ cluster_ingress_lookup.resources[0].spec.domain }}"
```

to something as:

```
- name: "cloudflare : Set ocp_hostname"
  set_fact:
    ocp_hostname: "router-default.{{ cluster_ingress_lookup.resources[0].spec.domain }}"
```

What you think of this change? Any suggestions?

**Comment by andrercm** [comment]
Thanks @QDespeisseTalan this is fixed in this PR: https://github.com/ibm-mas/ansible-devops/pull/1118

Now we're adding a new "A" type entry with cluster's ingress value:

<img width="1227" alt="image" src="https://github.com/ibm-mas/cli/assets/31037381/91d23745-f825-4475-ad59-d6f3a5bbb7d8">

MAS certificates seems to be generated just fine with that fix without any cert-manager controller workaround:

<img width="1401" alt="image" src="https://github.com/ibm-mas/cli/assets/31037381/5c5d46dc-5bb6-467c-8434-f099bfd3d1d8">
